frontend/src/Provider/ProviderComponents/ViewEditableTreatmentPlanModal.jsx

```diff
@@ -0,0 +1,50 @@
+import React, {useState} from "react";
+import Modal from "react-bootstrap/Modal"
+import { treatments } from "../../../../backend/data";
+
+
+export default function ViewTreatmentPlanModal({ patientId, onHide, show }) {
+    const treatment = treatments[patientId] // Simulate API call to retrieve the treatment plan of the patient
```

Comment:
To better simulate API calls, you can have `async` functions in a mock `api.js` class that just return the data in `treatments`. You can do this for everywhere you want to simulate an API call, as it will force you to make that code async / use Promises.

This way you would have to setup the JSX flow using async functions instead of indexing constants, which will match the final flow more. This would also require some changes to your hooks & state usage for these components.

```
async const getTreatments = (patientId) => {
    return treatments[patientId]
}
```